pkg/container/resolver_test.go

```diff
@@ -104,8 +111,88 @@ func TestResolverFail(t *testing.T) {
 		common.ToPtr(""),
 		common.ToPtr(false),
 		false,
+		nil,
 	})
 	specs, err = resolver.Finish()
 	assert.Error(t, err)
 	assert.Len(t, specs, 0)
 }
+
+func TestResolverLocalManifest(t *testing.T) {
+	currentUser, err := user.Current()
+	if err != nil {
+		t.Fatalf("Failed to get current user: %v", err)
+	}
+
+	if currentUser.Uid != "0" {
+		t.Skip("User is not root, skipping test")
+	}
+
+	_, err = exec.LookPath("podman")
+	if err != nil {
+		t.Skip("Podman not available, skipping test")
+	}
+
+	containerFile, err := os.CreateTemp("", "Containerfile")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer os.Remove(containerFile.Name())
+
+	tmpStorage, err := os.MkdirTemp("", "storage")
+	if err != nil {
+		t.Fatalf("Failed to create temp storage: %v", err)
+	}
+	defer os.Remove(tmpStorage)
+
+	// we need something more than scratch otherwise
+	// two containers with different arches will share
+	// the same underlying container image
+	_, err = containerFile.Write([]byte("FROM fedora-minimal"))
+	if err != nil {
+		t.Fatalf("Failed to create Containerfile: %v", err)
+	}
+
+	cmd := exec.Command( //nolint:gosec
+		"podman",
+		"--root", tmpStorage, // don't dirty the default store
+		"build",
+		"--platform", "linux/amd64,linux/arm64",
+		"--manifest", "multi-arch",
+		"-f", containerFile.Name(),
+		".",
+	)
+
+	err = cmd.Run()
+	if err != nil {
+		t.Fatalf("Error building podman containers: %v", err)
+	}
+
+	// try resolve an x86_64 container using a local manifest list
+	resolver := container.NewResolver("amd64")
+	resolver.Add(container.SourceSpec{
+		"localhost/multi-arch",
+		"",
+		common.ToPtr(""),
+		common.ToPtr(false),
+		true,
```

Comment:
(niptick) it feels slightly easier to read to have a local var `local:=true` and pass it here instead of `true` - this is my personal pet-peeve about APIs that take booleans as inputs, it's hard to know what the "true"/"false" stand for without additional context :)